app/main.py

```diff
@@ -1,6 +1,15 @@
-from typing import Callable
+from collections.abc import Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    def wrapper(*args) -> Callable:
+        if args in storage:
+            print("Getting from cache")
+        else:
+            print("Calculating new result")
+            storage.setdefault(args, func(*args))
```

Comment:
There is no more sense use `setdefault` method, because you are getting value at the end, we can simply set value by a key. Now it's doing one extra move - return you a value into nothing, so we don't need it.